app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class RecipesController < ApplicationController
+
+  def new; end #root
+
+  def index
+    recipes = EdamamApiWrapper.list_recipes(params[:query])
+
+    @recipes = Kaminari.paginate_array(recipes).page(params[:page]).per(12)
+
+  end
+
+  def show
+    @recipe = EdamamApiWrapper.display_recipe(params[:uri])
+  end
```

Comment:
You use `resources` for this route, which means that you're looking for `params[:id]`, not `params[:uri]`.